lib/timeline/component/item/BoxItem.js

```diff
@@ -126,41 +120,107 @@ BoxItem.prototype.redraw = function() {
-    dom.box.className = 'vis-item vis-box' + className;
-    dom.line.className = 'vis-item vis-line' + className;
-    dom.dot.className  = 'vis-item vis-dot' + className;
-
-    // set initial position in the visible range of the grid so that the
-    // rendered box size can be determinated correctly, even the content
-    // has a dynamic width (fixes #2032).
-    var previousRight = dom.box.style.right;
-    var previousLeft = dom.box.style.left;
-    if (this.options.rtl) {
-      dom.box.style.right = "0px";
-    } else {
-      dom.box.style.left = "0px";
-    }
-
-    // recalculate size
-    this.props.dot.height = dom.dot.offsetHeight;
-    this.props.dot.width = dom.dot.offsetWidth;
-    this.props.line.width = dom.line.offsetWidth;
-    this.width = dom.box.offsetWidth;
-    this.height = dom.box.offsetHeight;
+    this.dom.box.className = 'vis-item vis-box' + className;
+    this.dom.line.className = 'vis-item vis-line' + className;
+    this.dom.dot.className  = 'vis-item vis-dot' + className;
+  }
+}
 
-    // restore previous position
-    if (this.options.rtl) {
-      dom.box.style.right = previousRight;
-    } else {
-      dom.box.style.left = previousLeft;
+BoxItem.prototype._getDomComponentsSizes = function() {
+  return {
+    previous: {
+      right: this.dom.box.style.right,
+      left: this.dom.box.style.left
+    },
+    dot: {
+      height: this.dom.dot.offsetHeight,
+      width: this.dom.dot.offsetWidth
+    },
+    line: {
+      width: this.dom.line.offsetWidth
+    },
+    box: {
+      width: this.dom.box.offsetWidth,
+      height: this.dom.box.offsetHeight
     }
+  }
+}
+
+BoxItem.prototype._updateDomComponentsSizes = function(sizes) {
+  if (this.options.rtl) {
+    this.dom.box.style.right = "0px";
+  } else {
+    this.dom.box.style.left = "0px";
+  }
 
-    this.dirty = false;
+  // recalculate size
+  this.props.dot.height = sizes.dot.height;
+  this.props.dot.width = sizes.dot.width;
+  this.props.line.width = sizes.line.width;
+  this.width = sizes.box.width;
+  this.height = sizes.box.height;
+
+  // restore previous position
+  if (this.options.rtl) {
+    this.dom.box.style.right = sizes.previous.right;
+  } else {
+    this.dom.box.style.left = sizes.previous.left;
   }
 
-  this._repaintOnItemUpdateTimeTooltip(dom.box);
+  this.dirty = false;
+}
+
+BoxItem.prototype._repaintDomAdditionals = function() {
+  this._repaintOnItemUpdateTimeTooltip(this.dom.box);
   this._repaintDragCenter();
-  this._repaintDeleteButton(dom.box);
+  this._repaintDeleteButton(this.dom.box);
+}
+
+/**
+ * Repaint the item
+ * @param {boolean} [returnQueue=false]  return the queue
+ * @return {boolean} the redraw queue if returnQueue=true
+ */
+BoxItem.prototype.redraw = function(returnQueue) {
+  var sizes
+  var queue = [
+    // create item DOM
+    this._createDomElement.bind(this),
+
+    // append DOM to parent DOM
+    this._appendDomElement.bind(this),
+
+    // update dirty DOM. An item is marked dirty when:
```

Comment:
No, not comment here. I really think that placing it with the method definition is better.